medmodels/matching/metrics.py

```diff
@@ -63,6 +67,49 @@ def exact_metric(vector1: NDArray[np.float64], vector2: NDArray[np.float64]) ->
     return np.inf
 
 
+def mahalanobis_metric(
+    vector1: NDArray[np.float64],
+    vector2: NDArray[np.float64],
+    inv_cov: NDArray[np.float64],
```

Comment:
Based on the Mahalanobis formula, I would expect the covariance Matrix as input, so that the calculating of the inverse Covariance Matrix happens in the mahalanobis function and not before.